pandas/core/reshape.py

```diff
@@ -61,7 +61,7 @@ class _Unstacker(object):
     """
 
     def __init__(self, values, index, level=-1, value_columns=None,
-                 fill_value=None):
+                 fill_value=None, frame_constr=DataFrame):
```

Comment:
I could not think of a better way to make Unstacker return subclassed DataFrame, but am of course open to suggestions. ;-)

In either case, I have changed the variable name to return_type.